src/webgpu/api/validation/createBindGroup.spec.ts

```diff
@@ -793,3 +794,66 @@ g.test('buffer,usage')
       });
     }, !isValid);
   });
+
+g.test('buffer,resource_offset')
+  .desc(
+    `
+    Test that the resource.offset of the BindGroup entry is a multiple of limits.
+    'minUniformBufferOffsetAlignment|minStorageBufferOffsetAlignment' if the BindGroup entry defines
+    buffer and the buffer type is 'uniform|storage|read-only-storage'.
+  `
+  )
+  .params(u =>
+    u //
+      .combine('type', ['uniform', 'storage', 'read-only-storage'] as const)
+      .beginSubcases()
+      .expand('offset', ({ type }) =>
+        type === 'uniform'
+          ? [
+              kLimitInfo.minUniformBufferOffsetAlignment.default,
+              kLimitInfo.minUniformBufferOffsetAlignment.default * 2,
+              kLimitInfo.minUniformBufferOffsetAlignment.default * 3,
+              kLimitInfo.minUniformBufferOffsetAlignment.default + 20,
+            ]
+          : [
+              kLimitInfo.minStorageBufferOffsetAlignment.default,
+              kLimitInfo.minStorageBufferOffsetAlignment.default * 2,
+              kLimitInfo.minStorageBufferOffsetAlignment.default * 3,
+              kLimitInfo.minStorageBufferOffsetAlignment.default + 20,
+            ]
```

Comment:
We should test a value which is "minimally unaligned", that is, aligned to the largest power of 2 less than the required alignment. Which you can test with a case that does `* 0.5` or `* 1.5`.

Probably this is more useful than the `* 3` and `+ 20` cases so you could replace them, that is up to you.

Reply:
Ah, I didn't know it well. Thanks. I update the test to use `* 0.5` and `* 1.5`. Additionally, I also use `kBufferBindingTypes` from capability_info.ts here.